File: 171.excel-sheet-column-number.py
#
# @lc app=leetcode id=171 lang=python3
#
# [171] Excel Sheet Column Number
#

# @lc code=start
class Solution:
    def titleToNumber(self, s: str) -> int:
        size = len(s)
        pow_of_26 = 1
        num = 0
        for i in range(size-1, -1, -1):
            num = num + pow_of_26 * (ord(s[i]) - ord("A") + 1)
            pow_of_26 = pow_of_26 * 26
        return num

# @lc code=end

File: code/171.excel-sheet-column-number.cpp
/*
 * @lc app=leetcode id=171 lang=cpp
 *
 * [171] Excel Sheet Column Number
 */

#include <string>
#include <cmath>
using namespace std;

class Solution {
public:
    int titleToNumber(string s) {
        int size = s.size();
        int num = 0;
        for(int i = 0;i < size;i++) {
            num += (pow(26, size - i - 1) * int(s[i] - 'A' + 1));
        }
        return num;
    }
};

